### PR TITLE
[Enhancement] Fix some implicit conversions

### PR DESCRIPTION
**Pull request by zaorangyang**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

File: be/src/storage/convert_helper.cpp [hunk 1]
```diff
@@ -1077,10 +1078,10 @@ class DecimalToPercentileTypeConverter : public MaterializeTypeConverter {
                 dst_col->append_datum(dst_datum);
                 continue;
             }
-            double origin_value;
+            float origin_value;
             auto v = src_datum.get<CppType>();
             auto scale_factor = get_scale_factor<CppType>(src_type->scale());
-            DecimalV3Cast::to_float<CppType, double>(v, scale_factor, &origin_value);
```
Comment (trueeyu) [inline be/src/storage/convert_helper.cpp:1083]: Maybe have some problem here, please contact @satanson

File: be/src/storage/convert_helper.cpp [hunk 2]
```diff
@@ -1896,7 +1897,7 @@ Status RowConverter::init(const Schema& in_schema, const Schema& out_schema) {
 }
 
 void RowConverter::convert(std::vector<Datum>* dst, const std::vector<Datum>& src) const {
-    int num_datums = src.size();
+    int num_datums = static_cast<uint16_t>(src.size());
```
Comment (trueeyu) [inline be/src/storage/convert_helper.cpp:1900]: static_cast<int>

**Comment by trueeyu** [comment]
image

File: be/src/storage/convert_helper.cpp (hunk 2 above)
Comment (trueeyu) [reply be/src/storage/convert_helper.cpp:1900]: Need to be more careful to prevent this stupid bug

File: be/src/storage/chunk_helper.cpp [hunk 3]
```diff
@@ -261,7 +261,7 @@ std::vector<size_t> ChunkHelper::get_char_field_indexes(const vectorized::Schema
 
 void ChunkHelper::padding_char_columns(const std::vector<size_t>& char_column_indexes, const vectorized::Schema& schema,
                                        const starrocks::TabletSchema& tschema, vectorized::Chunk* chunk) {
-    size_t num_rows = chunk->num_rows();
+    uint32_t num_rows = static_cast<uint32_t>(chunk->num_rows());
```
Comment (chaoyli) [inline be/src/storage/chunk_helper.cpp:264]: if new_offset is an std::vector<uint32_t>, new_offset.resize(size_t), the conversion error will throw?;

File: be/src/storage/chunk_helper.cpp [hunk 4]
```diff
@@ -290,7 +290,7 @@ void ChunkHelper::padding_char_columns(const std::vector<size_t>& char_column_in
         }
 
         for (size_t j = 1; j <= num_rows; ++j) {
-            new_offset[j] = len * j;
+            new_offset[j] = static_cast<unsigned int>(len * j);
```
Comment (trueeyu) [inline be/src/storage/chunk_helper.cpp:293]: static_cast<uint32_t>

**Comment by trueeyu** [comment]
run starrocks_clang-tidy

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Review by trueeyu — DISMISSED** [review]

**Review by chaoyli — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by zaorangyang** [comment]
run starrocks_admit_test

**Comment by zaorangyang** [comment]
run starrocks_admit_test

**Comment by zaorangyang** [comment]
run starrocks_admit_test

**Comment by zaorangyang** [comment]
run starrocks_admit_test

**Comment by zaorangyang** [comment]
run starrocks_admit_test

**Comment by satanson** [comment]
rebase on the latest main branch to pass admit test

**Comment by zaorangyang** [comment]
run starrocks_admit_test

**Comment by zaorangyang** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by chaoyli — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: be/src/storage/convert_helper.cpp [hunk 5]
```diff
@@ -1746,7 +1747,7 @@ class DecimalFieldConverter : public FieldConverter {
                 return dst;
             }
         }
-        uint16_t num_items = src.size();
+        uint16_t num_items = static_cast<uint16_t>(src.size());
```
Comment (trueeyu) [inline be/src/storage/convert_helper.cpp:1750]: int num_items?

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by zaorangyang** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Review by chaoyli — DISMISSED** [review]

File: be/src/storage/convert_helper.cpp [hunk 6]
```diff
@@ -1311,7 +1312,7 @@ class DateToDateV2FieldConverter : public FieldConverter {
     ColumnPtr copy_convert(const Column& src) const override {
         auto nullable = src.is_nullable();
         auto dst = ChunkHelper::column_from_field_type(OLAP_FIELD_TYPE_DATE_V2, nullable);
-        uint16_t num_items = src.size();
+        uint16_t num_items = static_cast<uint16_t>(src.size());
         dst->reserve(num_items);
         for (int i = 0; i < num_items; ++i) {
```
Comment (trueeyu) [inline be/src/storage/convert_helper.cpp:1317]: `i` is int, `num_items` is uint16_t

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

File: be/src/storage/convert_helper.cpp [hunk 7]
```diff
@@ -1371,7 +1372,7 @@ class DateV2ToDateFieldConverter : public FieldConverter {
     ColumnPtr copy_convert(const Column& src) const override {
         auto nullable = src.is_nullable();
         auto dst = ChunkHelper::column_from_field_type(OLAP_FIELD_TYPE_DATE, nullable);
-        uint16_t num_items = src.size();
+        uint16_t num_items = static_cast<uint16_t>(src.size());
         dst->reserve(num_items);
         for (int i = 0; i < num_items; ++i) {
```
Comment (trueeyu) [inline be/src/storage/convert_helper.cpp:1377]: i is int, num_items is uint16_t

File: be/src/storage/convert_helper.cpp [hunk 8]
```diff
@@ -1311,7 +1312,7 @@ class DateToDateV2FieldConverter : public FieldConverter {
     ColumnPtr copy_convert(const Column& src) const override {
         auto nullable = src.is_nullable();
         auto dst = ChunkHelper::column_from_field_type(OLAP_FIELD_TYPE_DATE_V2, nullable);
-        uint16_t num_items = src.size();
+        int num_items = src.size();
```
Comment (trueeyu) [inline be/src/storage/convert_helper.cpp:1315]: static_cast<int>, size_t --> int

**Review by chaoyli — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by trueeyu — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Review by chaoyli — APPROVED** [review]

**Pull request merged**